src/Yaml/test.zig

```diff
@@ -113,6 +113,60 @@ test "several integer bases" {
     try testing.expectEqualSlices(i8, &[_]i8{ 10, -10, 16, -16, 8, -8 }, &arr);
 }
 
+test "simple flow sequence / bracket list" {
+    const source =
+        \\a_key: [a, b, c]
+    ;
+
+    var yaml: Yaml = .{ .source = source };
+    defer yaml.deinit(testing.allocator);
+    try yaml.load(testing.allocator);
+
+    try testing.expectEqual(yaml.docs.items.len, 1);
+
+    const map = yaml.docs.items[0].map;
+
+    const list = map.get("a_key").?.list;
+    try testing.expectEqual(list.len, 3);
+
+    try testing.expectEqualStrings("a", list[0].string);
+    try testing.expectEqualStrings("b", list[1].string);
+    try testing.expectEqualStrings("c", list[2].string);
+}
+
+test "simple flow sequence / bracket list with trailing comma" {
+    const source =
+        \\a_key: [a, b, c,]
+    ;
+
+    var yaml: Yaml = .{ .source = source };
+    defer yaml.deinit(testing.allocator);
+    try yaml.load(testing.allocator);
+
+    try testing.expectEqual(yaml.docs.items.len, 1);
+
+    const map = yaml.docs.items[0].map;
+
+    const list = map.get("a_key").?.list;
+    try testing.expectEqual(list.len, 3);
+
+    try testing.expectEqualStrings("a", list[0].string);
+    try testing.expectEqualStrings("b", list[1].string);
+    try testing.expectEqualStrings("c", list[2].string);
+}
+
+test "simple flow sequence / bracket list with invalid comment" {
+    const source =
+        \\a_key: [a, b, c]#invalid
```

Comment:
It looks like this *is* a valid YAML according to several parses available online such as [this one](https://yamlchecker.com/). However, this should fail I believe:

```yaml
a_key: [a, b, c,,]
```

Reply:
Interesting.
That same example will fail on https://www.yamllint.com/
`Comments must be separated from other tokens by white space characters at line 1, column 17`

Reading the spec, I think the second parser is right: https://yaml.org/spec/1.2.2/#66-comments
`Comments must be separated from other tokens by white space characters`

Now, given some parsers seem to accept it and others do not, I don't have a strong personal preference on whether or not adhering to that particular spec is worth it. 

Is the aim of this project spec compliance or compatibility with the broadest set of files which might be out there?

Reply:
I would prefer we stayed compliant with the spec 👍

Reply:
Sounds good, the current implementation is spec compliant then I think.